src/js/utils.js

```diff
@@ -322,6 +322,35 @@ function parseCookie(str, opts) {
   return parsed;
 }
 
+function CheckedDNTBuffer(size) {
+  if (size) {
+    this.size = size;
+  }
+  this.index = 0;
+  this.array = new Array(this.size);
+  this.set = new Set();
+}
+
+CheckedDNTBuffer.prototype = {
+  size: 1000,
+  // if we've seen item, return true. if we haven't store it and return false
+  hasOrSet: function(item) {
```

Comment:
I'm not sure misuse of the API ought to be a huge concern; making it idempotent is fairly light (considering it's a couple lines of code and a constant time operation). I guess my only thought then is whether an adequate name can be found that makes it clear what this is expected to do.

How about something like `containsOrInsert` or `containsOrAdd`?